policies.go

```diff
@@ -344,23 +344,28 @@ func (r *roundRobinHostPolicy) SetPartitioner(partitioner string)   {}
 func (r *roundRobinHostPolicy) Init(*Session)                       {}
 
 func (r *roundRobinHostPolicy) Pick(qry ExecutableQuery) NextHost {
-	// i is used to limit the number of attempts to find a host
-	// to the number of hosts known to this policy
-	var i int
-	return func() SelectedHost {
-		hosts := r.hosts.get()
-		if len(hosts) == 0 {
-			return nil
-		}
-
+	hosts := r.hosts.get()
+	var startIdx int
```

Comment:
I'd think it would be cleaner here to return `nil` immediately if `len(hosts) == 0`

Reply:
That's a stylistic change I'd be happy to do.

But are you sure? None of the existing Pick() implementations returns nil - the function they return can return nil, but the NextHost function itself is always non-nil.

I tried making the change, but one of the tests currently asserts that Pick() returns a non-nil function even when there are no hosts.

Reply:
Uhm, ok, that's a valid point. Ignore then ;-)